File: frontend/src/api/knowledge.ts
import { http } from './http'
export interface KnowledgeHit { id:string; content:string; title_path:string; document_id:string; version_id:string; chunk_index:number; score:number; metadata:Record<string, unknown> }
export interface SearchResponse { query:string; hits:KnowledgeHit[]; retrieval_mode:string; embedding_dimension:number }
export interface KnowledgeDocument { id:string; knowledge_base_id:string; document_key:string; file_name:string; source_path:string; file_type:string; file_size:number; content_hash:string; category:string; permission_level:string; status:string; created_at:string; updated_at:string }
export const knowledgeApi = {
  search: (query:string, top_k=5) => http.post<{data:SearchResponse}>('/knowledge/search',{query,top_k}),
  defaultBase: () => http.get('/knowledge/default'),
  list: () => http.get<{data:KnowledgeDocument[]}>('/knowledge/documents'),
  create: (payload: Partial<KnowledgeDocument>) => http.post<{data:KnowledgeDocument}>('/knowledge/documents', payload),
  delete: (id:string) => http.delete<{data:{document_id:string;status:string}}>(`/knowledge/documents/${id}`),
  ingest: (id:string) => http.post<{data:{document_id:string;status:string}}>(`/knowledge/documents/${id}/ingest`),
}

File: app/schemas/knowledge.py
from datetime import datetime
from typing import Any

from pydantic import BaseModel, Field


class KnowledgeBaseResponse(BaseModel):
    id: str
    name: str
    description: str
    status: str


class DocumentCreateRequest(BaseModel):
    knowledge_base_id: str | None = None
    document_key: str = Field(min_length=1, max_length=1000)
    file_name: str = Field(min_length=1, max_length=255)
    source_path: str = Field(min_length=1, max_length=1000)
    file_type: str = Field(min_length=1, max_length=32)
    file_size: int = Field(default=0, ge=0)
    content_hash: str = Field(min_length=64, max_length=64)
    category: str = Field(default="general", max_length=64)
    permission_level: str = Field(default="public", max_length=32)


class DocumentIngestResponse(BaseModel):
    document_id: str
    status: str
    event_id: str | None = None


class DocumentResponse(BaseModel):
    id: str
    knowledge_base_id: str
    document_key: str
    file_name: str
    source_path: str
    file_type: str
    file_size: int
    content_hash: str
    category: str
    permission_level: str
    status: str
    created_at: datetime
    updated_at: datetime


class KnowledgeSearchRequest(BaseModel):
    query: str = Field(min_length=1, max_length=4000)
    top_k: int = Field(default=5, ge=1, le=50)
    knowledge_base_id: str | None = None
    category: str | None = None


class KnowledgeSearchHit(BaseModel):
    id: str
    content: str
    title_path: str
    document_id: str
    version_id: str
    chunk_index: int
    score: float
    metadata: dict[str, Any] = Field(default_factory=dict)


class KnowledgeSearchResponse(BaseModel):
    query: str
    hits: list[KnowledgeSearchHit]
    retrieval_mode: str
    embedding_dimension: int
